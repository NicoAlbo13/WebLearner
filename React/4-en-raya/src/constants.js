export const TURNS = {
  p1: 'ðŸŸ¡',
  p2:'ðŸŸ¢'
}

export const COLORS = {
  c1:'#FFFF00',
  c2:'#00FF00'
}