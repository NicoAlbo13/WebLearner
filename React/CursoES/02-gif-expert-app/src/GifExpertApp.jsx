import { useState } from "react"
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(["One Punch Man"]);

    const handleAddCategory = (newCategory)=>{
        if(categories.includes(newCategory)) return;
        setCategories([newCategory,...categories])
    }

    return (
    <>
        <h1>GifExpertApp</h1>

        <AddCategory onAddCategory={handleAddCategory}/>

        {categories.map((element)=>(
                <GifGrid key={element} category={element}/>
            )
        )}
    </>
    )
}
