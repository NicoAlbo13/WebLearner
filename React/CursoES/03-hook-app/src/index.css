body{
  padding: 20px;
}

button{
  margin-right: 12px;
}

a{
  margin-right: 12px;
}
