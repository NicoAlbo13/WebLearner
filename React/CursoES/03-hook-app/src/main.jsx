import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import './index.css'
import { MainApp } from './09-useContext/MainApp'
// import { MemoHook } from './06-memos/MemoHook'
// import { CallbackHook } from './06-memos/CallbackHook'
// import { Padre } from './07-tarea-memo/Padre'
// import HooksApp from './HooksApp'
// import { CounterApp } from './01-useState/CounterApp'
// import { CounterCustomHook } from './01-useState/CounterCustomHook'
// import { SimpleForm } from './02-useEffect/SimpleForm'
// import { SimpleFormCustomHook } from './02-useEffect/SimpleFormCustomHook'
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
// import { FocusScreen } from './04-useRef/FocusScreen'
// import { Layout } from './05-useLayoutEffect/Layout'
// import './08-useReducer/intro-reducer'
// import { TodoApp } from './08-useReducer/TodoApp'
import { BrowserRouter } from "react-router";

createRoot(document.getElementById('root')).render(
  // <StrictMode>
  <BrowserRouter>
    <MainApp/>
  </BrowserRouter>
  // </StrictMode>,
)
