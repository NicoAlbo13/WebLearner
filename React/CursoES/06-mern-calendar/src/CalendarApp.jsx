import { BrowserRouter } from "react-router"
import {Provider} from 'react-redux'
import { AppRouter } from "./routes/AppRouter"
import { store } from "./store/store"

export const CalendarApp = () => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <AppRouter/>
      </BrowserRouter>
    </Provider>
  )
}
