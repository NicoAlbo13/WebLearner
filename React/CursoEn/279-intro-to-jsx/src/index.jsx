import React from "react";
import ReactDOM from "react-dom";

const name = "albo";

ReactDOM.render(<h1 className="heading" contentEditable="true" spellCheck="false">Hello World {name}</h1>, document.getElementById("root"));
