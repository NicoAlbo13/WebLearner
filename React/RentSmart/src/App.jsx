import { useState } from 'react'
import Auth from './views/Auth'

function App() {

  return (
    <>
      <Auth/>
    </>
  )
}

export default App
