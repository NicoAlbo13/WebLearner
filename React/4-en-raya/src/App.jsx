import { useEffect, useState } from 'react';
import confetti from 'canvas-confetti'
import Square from './components/Square';
import WinnerDial from './components/WinnerDial';
import { checkEnd } from './logic/check';
import { checkWinner } from './logic/check';
import { TURNS } from './constants';
import { COLORS } from './constants';


function App() {
  const [turn, setTurn] = useState(TURNS.p1);
  const [board, setBoard]=useState(Array(42).fill(null));
  const [winner, setWinner] = useState(null);//null is not a winner, false is tie

  function resetGame(){ 
    setTurn(TURNS.p1);
    setBoard(Array(42).fill(null))
    setWinner(null);
    setColor(COLORS.c1)
  }

  function makeMove(index){

    if(board[index]!=null)return
    const newBoard = [...board];
    //Made all pieces start from bottom row
    let finalIndex=index;
    const row = Math.floor((finalIndex/7));
    for(let i=(5-row);i>0;i--){
      if(board[finalIndex+7]==null){
        finalIndex+=7;
      }
    }
    newBoard[finalIndex] = turn;
    setBoard(newBoard)

    const newTurn = (turn==TURNS.p1) ? TURNS.p2 : TURNS.p1;
    setTurn(newTurn)
    const newColor = (color==COLORS.c1) ? COLORS.c2 : COLORS.c1;
    setColor(newColor)  

    const newWinner = checkWinner(newBoard);
    //console.log("NewWinner",newWinner)
    if(newWinner){
      setEnable(!enable)
      confetti()
      setWinner(newWinner);
      //console.log("NewWinner inside", winner);
    }else if(checkEnd(newBoard)){
      setEnable(!enable)
      setWinner(false);
    }
  }

  const [enable, setEnable] = useState(false);
  const [position, setPosition] = useState({ x: 0, y: 0 });
  const [color, setColor] = useState(COLORS.c1);
  useEffect(()=>{
    const handleMove = (event) => {
      const { clientX, clientY } = event
      setPosition({ x: clientX, y: clientY })
    }

    if (enable) {
      document.addEventListener('pointermove', handleMove)
    }

    return ()=>{
      setPosition({ x: 0, y: 0 })
      document.removeEventListener('pointermove', handleMove)
    }
  },[enable])

  return (
    <>
      <div style={{
          position: 'absolute',
          backgroundColor: `${color}`,
          border: '4px solid black',
          borderRadius: '50%',
          opacity: 0.8,
          pointerEvents: 'none',
          left: -25,
          top: -25,
          width: 50,
          height: 50,
          transform: `translate(${position.x}px, ${position.y}px)`
        }}
        />
      <main className="board">
        <h1>Connect Four</h1>
        <button onClick={()=>{setEnable(!enable)}}>{enable?"Disable":"Enable"} Follower</button>
        <button onClick={resetGame}>Reset game</button>
        <section className='game'>
          {board.map((element, index)=>{
            //console.log(element);
            return(
            <Square 
            key={index}
            index={index}
            onClick={makeMove}
            element = {element}
            />
            )})}
        </section>
        <section className='turn'>
            <Square
              element={TURNS.p1}
              mark={turn===TURNS.p1}/>
            <Square
              element={TURNS.p2}
              mark={turn===TURNS.p2}/>
        </section>
        <WinnerDial winner={winner} resetGame={resetGame}/>
      </main>
    </>
  )
}

export default App
