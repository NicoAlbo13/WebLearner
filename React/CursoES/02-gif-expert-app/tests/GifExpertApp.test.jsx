import { fireEvent, render, screen } from '@testing-library/react';
import { GifExpertApp } from "../src/GifExpertApp";

describe('test on GifExpertApp component', () => {

    const newCategory = 'Dragon Ball';

    test('should match Snapshot', () => {
        const { container } = render(<GifExpertApp/>)
        expect(container).toMatchSnapshot();
    })

    test('should add a new category when submitted', () => {
        render(<GifExpertApp/>)
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form')

        fireEvent.input(input, {target:{value: newCategory}});
        fireEvent.submit(form);

        expect(screen.getAllByRole('heading', {level: 3}).length).toBe(2)//taking in count only h3 handle categories
        // screen.debug()
    })

    test('should not add repeated category', () => {
        render(<GifExpertApp/>)
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form')

        //add same value twice
        fireEvent.input(input, {target:{value: newCategory}});  
        fireEvent.input(input, {target:{value: newCategory}});
        fireEvent.submit(form);

        //only one should be added
        expect(screen.getAllByRole('heading', {level: 3}).length).toBe(2)//taking in count only h3 handle categories
    })

})
