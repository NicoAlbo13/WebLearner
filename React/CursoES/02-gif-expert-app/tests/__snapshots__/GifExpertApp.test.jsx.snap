// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on GifExpertApp component should match Snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search your Gif"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch Man
  </h3>
  <h2>
    Loading Images...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
